torch/csrc/distributed/c10d/TraceUtils.h

```diff
@@ -438,6 +455,7 @@ struct NCCLTraceBuffer {
       }
       if (completed) {
         r.state_ = "completed";
+        r.duration_ = getDurationFromFirstEvent(*r.start_, *r.end_);
```

Comment:
If timing is disabled, then `r.start_` can be nullptr here and it will crash. I think we might want to turn on timing and start events if the flight recorder is enabled. That way we don't have all these different pathways where some information is not gathered.

Reply:
yea, if possible lets enable timing by default. but i should make this safe regardless.  (IIRC there are checks _inside_ the getDuration fn but i probably forgot about the * or thought that it was always safe to deref the first layer.